### PR TITLE
🚧(Backend) : Models and Repositories

### LINKED ISSUE #34: Create models and repositories
Create all db models in Spring-boot
Create the structure of repositories in Spring-boot

### PR DESCRIPTION
**Pull request by priyo5**
Created Data models for collections in db
Created Repositories for Query structures 
Closes #34

### CODE REVIEW

File: Server/src/main/java/Database/TheAuPair/Models/Activity.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    {
+        return budget;
+    }
+
+    public void setBudget(double budget)
+    {
+        this.budget = budget;
+    }
+
+    public String getCom()
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Models/Activity.java:116]: What does this value refer to? Can it maybe be more descriptive

File: Server/src/main/java/Database/TheAuPair/Models/Activity.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    @Id
+    private String id;
+
+    @Field ("name")
+    private String name;
+    @Field ("description")
+    private String description;
+    @Field ("location")
+    private String loc;
+    @Field ("time_start")
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Models/Activity.java:19]: Should we not make this CamelCase to keep consistency? Unless it's generated by the DB or something?

File: Server/src/main/java/Database/TheAuPair/Models/Au_Pair.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+@Document("Au-Pair")
+public class Au_Pair
+{
+    @Id
+    private String id;
+
+    @Field("rating")
+    private double rating;
+    @Field("pay_rate")
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Models/Au_Pair.java:15]: Same here,  please confirm

File: Server/src/main/java/Database/TheAuPair/Models/Med_Aid.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+@Document("Medical_Aid")
+public class Med_Aid
+{
+    @Id
+    private String id;
+
+    @Field ("plan")
+    private String plan;
+    @Field ("main_name")
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Models/Med_Aid.java:15]: Same here

File: Server/src/main/java/Database/TheAuPair/Models/Child.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
+package Database.TheAuPair.Models;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Models/Child.java:6]: empty line

File: Server/src/main/java/Database/TheAuPair/Models/Parent.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
+package Database.TheAuPair.Models;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Models/Parent.java:6]: empty line

File: Server/src/main/java/Database/TheAuPair/Repositories/ActivityRepository.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package Database.TheAuPair.Repositories;
+
+import Database.TheAuPair.Models.Activity;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.mongodb.repository.Query;
+
+
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Repositories/ActivityRepository.java:7]: blank line

File: Server/src/main/java/Database/TheAuPair/Repositories/UserRepository.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package Database.TheAuPair.Repositories;
+
+import Database.TheAuPair.Models.User;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.mongodb.repository.Query;
+
+
```
Comment (Ruben-Brits) [inline Server/src/main/java/Database/TheAuPair/Repositories/UserRepository.java:7]: blank line

**Review by Ruben-Brits — CHANGES_REQUESTED** [review]
Just 2 small things and whitespace then all good

**Comment by priyo5** [comment]
Changes will be made to update this PR
- Camel case will be adhered to (changes to db to be made to allow this)
- Autogen functions are currently present will update them to a more meaningful nameing scheme
- blank lines to be removed

File: Server/src/main/java/Database/TheAuPair/Models/Au_Pair.java (hunk 3 above)
Comment (d-azmanov) [reply Server/src/main/java/Database/TheAuPair/Models/Au_Pair.java:15]: This is the pay rate (hourly rate) that an au pair will set up for their account which will be used to determine the au pairs salary.  I think this is fine.

**Review by Ruben-Brits — APPROVED** [review]

**Pull request merged**